lib/item.py

```diff
@@ -391,17 +405,20 @@ def _castvalue_to_itemtype(self, value, compat):
         """
         # casting of value, if compat = latest
         if compat == ATTRIB_COMPAT_LATEST:
-            mycast = globals()['_cast_' + self._type]
-            try:
-                value = mycast(value)
-            except:
-                logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type, str(mycast)))
-                if isinstance(value, list):
-                    value = []
-                elif isinstance(value, dict):
-                    value = {}
-                else:
-                    value = mycast('')
+            if self._type != None:
+                mycast = globals()['_cast_' + self._type]
+                try:
+                    value = mycast(value)
+                except:
+                    logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type, str(mycast)))
+                    if isinstance(value, list):
+                        value = []
+                    elif isinstance(value, dict):
+                        value = {}
+                    else:
+                        value = mycast('')
+            else:
+                logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type))
```

Comment:
I think it is helpful. Otherwise the user would not recognize that he tries to assign a value to a var which has no type. It is not, that the type is wrong. The user simply has not defined a type for that var. On the other hand to clarify where the message comes from, I wanted it to be the same as with the other cast-errors that may occur.